stringbasic/String.cpp

```diff
@@ -0,0 +1,58 @@
+#include "String.h"
+
+#include <cstring>
+#include <iostream>
+
+String::String()
+{}
+
+String::String(const char* string) {
+	changeString(string);
+}
+
+String::String(const String& other) {
+	(*this) = other;
+}
+
+String& String::operator=(const String& other) {
+	if (this == &other) {
+		return *this;
+	}
+
+	changeString(other.m_string_ptr);
```

Comment:
Similar to your copy CTOR, there's a double free bug in here also.
Make sure the underlying resource (aka the pointer and the memory it points to) is managed by one and one entity only.